dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/api/FileReport.java

```diff
@@ -0,0 +1,44 @@
+package edu.harvard.iq.dataverse.api;
+
+import edu.harvard.iq.dataverse.dataset.DatasetReportService;
+import edu.harvard.iq.dataverse.persistence.user.AuthenticatedUser;
+
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.StreamingOutput;
+import java.time.LocalDateTime;
+
+@Path("reports/file")
+public class FileReport extends AbstractApiBean {
+
+    @Inject
+    private DatasetReportService datasetReportService;
+
+    // -------------------- LOGIC --------------------
+
+    @Path("/")
+    @GET
+    @Produces(MediaType.APPLICATION_OCTET_STREAM)
+    public Response createFileReport() {
+        try {
+            AuthenticatedUser user = findAuthenticatedUserOrDie();
```

Comment:
If we want to limit this call to superusers only service method should be safe as well (i.e: since we cannot annotate for this specific 'permission' we should make explicit check in the service if user calling it is a superuser). For example check `Users`. 

NOTE: Exceptions thrown from service will be wrapped in EJBException.

Reply:
I'm not convinced about duplicating this permission check in the service. One reason is that it would spoil the existing report building code with some "alien" (unrelated) logic. And the second is that it's going to be used only by the api call, and accidental using it elsewhere is not very probable, as it would require creating some stream and redirecting it to the user.

Reply:
I didn't mean to duplicate but to move this logic to the service. I know it's 'alien' but we have no better way of insuring this with annotation so you know: better safe than sorry ;) 
But I'll leave it up to you.

Reply:
In fact we have to duplicate it, as we are updating the api use time for the user.
Ok, I think that it would be best to discuss it tomorrow during daily call.